src/background.ts

```diff
@@ -244,9 +245,15 @@ omnibox.init()
 
 // }}}
 
-browser.commands.onCommand.addListener((command_name: string) => {
-    if (command_name !== "escape_hatch") return
-    excmds_background.escapehatch()
+browser.commands.onCommand.addListener(async (command_name: string) => {
+    const commands = await browser.commands.getAll()
+    const command = commands.filter(c => c.name == command_name)[0]
+    const exstring = config.get(
+        "browsermaps",
+        keyseq.mozMapToMinimalKey(command.shortcut).toMapstr(),
+    )
+    if (exstring in useractions) return useractions[exstring]()
```

Comment:
Even if I comment out all of the other stuff in this function, and remove the async-yness, just leaving
`(command_name: string) => useractions.escapehatch()` Firefox still thinks we aren't in a user action.

: (

Reply:
If `onCommand` isn't currently considered a user action I think Mozilla would be willing to change that.

Reply:
It is. The problem must be this:

> Also, if a user input handler waits on a promise, then its status as a user input handler is lost.

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/User_actions

We could just open/close the sidebar every time synchronously and then do the other stuff, except we can't even await the sidebar opening / closing.